src/dmd/expressionsem.d

```diff
@@ -10225,6 +10225,24 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return setError();
         }
 
+        if (t1.ty == Tclass && t2.ty == Tclass)
+        {
+            // Lower to object.__cmp(e1, e2)
+            Expression cl = new IdentifierExp(exp.loc, Id.empty);
+            cl = new DotIdExp(exp.loc, cl, Id.object);
+            cl = new DotIdExp(exp.loc, cl, Id.__cmp);
```

Comment:
As discussed on other PRs: don't use __ as it can lead to weird side effects - use `_d_`

Reply:
`__cmp` is already defined in druntime and it’s already used in the lowering of array comparisons. This PR just uses an overload of it.

Reply:
@wilzbach In this case the `Id.__cmp` is unavoidable.  It's the same for the `__ArrayCast` identifier that I refactored in #9572.  We'd have to refactor `id.d` and object.d to correct the issue.  See how they are mixed in here:  https://github.com/JinShil/dmd/blob/master/src/dmd/id.d#L26-L42

If this were a new symbol, it would be prudent to have it changed, but since this already exists, I think it would be too disruptive to include it in this PR.  It could be remedied in a separate DMD and druntime PR, however.